sematic/api/endpoints/artifacts.py

```diff
@@ -1,19 +1,28 @@
 # Standard Library
-from typing import List
+from http import HTTPStatus
+from typing import List, Optional
 
 # Third-party
 import flask
 import sqlalchemy
+from sqlalchemy.orm.exc import NoResultFound
 
 # Sematic
 from sematic.api.app import sematic_api
-from sematic.api.endpoints.request_parameters import get_request_parameters
+from sematic.api.endpoints.auth import authenticate
+from sematic.api.endpoints.request_parameters import (
+    get_request_parameters,
+    jsonify_error,
+)
 from sematic.db.db import db
 from sematic.db.models.artifact import Artifact
+from sematic.db.models.user import User
+from sematic.db.queries import get_artifact
 
 
 @sematic_api.route("/api/v1/artifacts", methods=["GET"])
-def list_artifacts_endpoint() -> flask.Response:
+@authenticate
```

Comment:
Maybe we can make it so we have to mark when something *shouldn't* authenticate, rather than having to mark when it should? The common-case is authenticated, and we're also less likely to have security holes if we go with the auth-required-by-default approach. This is not a comment for fixing in this PR; if you agree just file a ticket for later.

Reply:
I think flask has hooks that allow you to apply certain kinds of middleware stuff to all requests. We can probably leverage something of that nature. anyway, yeah, for another time!